torch/fx/experimental/normalize.py

```diff
@@ -0,0 +1,77 @@
+import torch
+import torch.fx
+import inspect
+from typing import Any, Dict, Optional, Tuple
+from torch.fx.node import Argument, Target
+from torch._jit_internal import boolean_dispatched
+
+from torch.fx import Transformer
+
+class NormalizeArgs(Transformer):
+    """
+    Normalize arguments to Python targets. This means that
+    `args/kwargs` will be matched up to the module/functional's
+    signature and rewritten to exclusively kwargs in positional order.
+    Also populates default values. Does not support positional-only
+    parameters or varargs parameters (*args, **kwargs).
+    """
+    def __init__(self, module : torch.nn.Module, normalize_functionals : bool = True,
+                 normalize_modules : bool = True):
+        super().__init__(module)
+        self.normalize_functionals = normalize_functionals
+        self.normalize_modules = normalize_modules
+
+    def call_function(self, target : Target, args : Tuple[Argument, ...], kwargs : Dict[str, Any]):
+        if self.normalize_functionals and target.__module__ == 'torch.nn.functional':
+            target_for_analysis = target
+            if target in boolean_dispatched:
+                # HACK: `boolean_dispatch` as used in `torch.nn.functional` makes it so that we have
```

Comment:
🤔 Why does this exist lol.

Reply:
Because TorchScript (of course lol) https://github.com/pytorch/pytorch/pull/14425